odoo/addons/base/models/ir_ui_menu.py

```diff
@@ -93,15 +93,31 @@ def _visible_menu_ids(self, debug=False):
 
         # process action menus, check whether their action is allowed
         access = self.env['ir.model.access']
-        MODEL_GETTER = {
-            'ir.actions.act_window': lambda action: action.res_model,
-            'ir.actions.report': lambda action: action.model,
-            'ir.actions.server': lambda action: action.model_id.model,
+        MODEL_FIELD = {
```

Comment:
That name is not clear to me.  If we call `action._name` its "category" or "type", then I suggest `MODEL_BY_CATEGORY` or `MODEL_BY_TYPE`.